### PR TITLE
Tweaks to Style/ConditionalAssignment 

### PR DESCRIPTION
**Pull request by rrosenblum**
~~This is designed to be merged in after #2433. It could also replace that PR entirely.~~

### CODE REVIEW

File: spec/rubocop/cop/style/conditional_assignment_spec.rb [hunk 1]
```diff
@@ -1,12 +1,15 @@
 # encoding: utf-8
 
 require 'spec_helper'
+require 'pry'
```
Comment (jonas054) [inline spec/rubocop/cop/style/conditional_assignment_spec.rb]: I think you left this line here by mistake.

Comment (rrosenblum) [inline spec/rubocop/cop/style/conditional_assignment_spec.rb]: Thanks for catching that. I constantly forget to remove that require.

**Comment by rrosenblum** [comment]
I finally finished the rebase. It was probably one of the most difficult rebases that I have ever had to deal with. I found a couple of edge cases that I need to cover. Also, after looking at some of the auto-corrections, I am going to simplify the default default configuration, and I am going to only register an offense for single line branches. I originally thought extracting the assignment could help emphasize the assignment. I have seen enough cases to prove that this is not always true.

@alexdowad some of your ideas really helped me to clean up this code.

**Comment by alexdowad** [comment]
Thanks @rrosenblum.

**Comment by bbatsov** [comment]
There are still some conflicts with `master`.

**Comment by rrosenblum** [comment]
@bbatsov I think I finally have everything completed how I would like it. I have renamed the configuration option, made the cop work with ternary operators, and made the cop work with comparable methods.

**Comment by bbatsov** [comment]
The build is failing due to:

```
lib/rubocop/cop/style/conditional_assignment.rb:223:18: C: Avoid using nested modifiers.
          return if branches.any?(&:begin_type?) if single_line_conditions_only?
```

P.S. We should improve this offence message a bit (e.g. by include the modifier name in it).

**Comment by bbatsov** [comment]
:+1:

**Comment by alexdowad** [comment]
:+1:

**Pull request merged**